### ISSUE TITLE
[BUG]Multi-node Install of harvester using IPXE Vagrant is failing on Provo DCP Cloud

### ISSUE THREAD

**Issue by chajain** [harvester/harvester#1148]
**Describe the bug**
When trying to bring up the multi-cluster Harvester Env using the instructions on https://github.com/harvester/ipxe-examples/tree/main/vagrant-pxe-harvester, the first node boots fine but subsequent nodes fail to install.

**To Reproduce**
1. Follow the README on https://github.com/harvester/ipxe-examples/tree/main/vagrant-pxe-harvester and try to bring up a multi-cluster Harvester Env on one of the nodes on DCP Cloud. From syslogs, looks like its getting stuck in this loop.
Sep 01 23:45:30 cjain-leap.novalocal rke2[2463]: time="2021-09-01T23:45:30Z" level=info msg="Waiting to retrieve agent configuration; server is not ready: Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] Processing secret custom-c54ba355db1a-machine-plan in namespace fleet-local at generation 0 with resource version 6989"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] Calculated checksum to be 046ed33e020f2a69379c6501ffd2655dcc8b8d94972aab1efae54ef16597bc02"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] Remote plan had an applied checksum value of 046ed33e020f2a69379c6501ffd2655dcc8b8d94972aab1efae54ef16597bc02"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] Applied checksum was the same as the plan from remote. Not applying."
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] needsApplied was false, not applying"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] writing an applied checksum value of 046ed33e020f2a69379c6501ffd2655dcc8b8d94972aab1efae54ef16597bc02 to the remote plan"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] Enqueueing after 5.000000 seconds"
Sep 01 23:45:34 cjain-leap.novalocal rancher-system-agent[2259]: time="2021-09-01T23:45:34Z" level=debug msg="[K8s] secret data/string-data did not change, not updating secret"
Sep 01 23:45:34 cjain-leap.novalocal rke2[2463]: time="2021-09-01T23:45:34Z" level=info msg="Waiting to retrieve agent configuration; server is not ready: Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"

**Expected behavior**
All the nodes should install properly and discovered by Harvester UI

**Support bundle**

**Environment:**
 - Harvester ISO version: Master
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630):IPXE Vagrant 

**Additional context**

**Comment by guangyee** [comment]
Related to https://github.com/harvester/harvester/issues/1137

**Comment by bk201** [comment]
This should be fixed in the latest master ISO. The OS doesn't pull metadata from cloud providers now.

**Comment by lanfon72** [comment]
With https://github.com/harvester/ipxe-examples/pull/16 and [v0.3.0-preview1](https://github.com/harvester/harvester/releases/tag/v0.3.0-preview1), I can't reproduce this issue on my laptop environment.
(I think the issue is not strongly required to be reproduced on bare metals as it was using vagrant for the cluster.)

As @bk201 mentioned, I also tried installing nodes without external networking, it did trying to pull some **squashfs** file from our https://releases.rancher.com/ ,I think it's different to the related issue #1137 mentioned. (but if Provo DCP Cloud can't accees our release website, it should be the root cause.)

I would suggest to close this issue as the related issue #1137 has been resolved and verified.
Or I will try to reproduce it after I can access our Provo Machines. (should be in this week)
